references/cli/system.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/oam-dev/cluster-gateway/pkg/generated/clientset/versioned"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/oam-dev/kubevela/apis/types"
+	"github.com/oam-dev/kubevela/pkg/multicluster"
+	"github.com/oam-dev/kubevela/pkg/utils/common"
+)
+
+const (
+	// FlagDeploymentName specifies the deployment name
+	FlagDeploymentName = "name"
+)
+
+// NewSystemCommand print system detail info
+func NewSystemCommand(c common.Args) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "system",
+		Short: "Manage system.",
+		Long:  "Manage system, incluing printing the system deployment information in vela-system namespace and diagnosing the system's health.",
+		Example: "# Check all deployments information:\n" +
+			"> vela system info\n" +
+			"# Specify a deployment name to check detail information:\n" +
+			"> vela system info --name kubevela-vela-core\n" +
+			"# Diagnose the system's health:\n" +
+			"> vela system diagnose\n",
+		Annotations: map[string]string{
+			types.TagCommandType: types.TypeSystem,
+		},
+	}
+	cmd.AddCommand(
+		NewSystemInfoCommand(c),
+		NewSystemDiagnoseCommand(c))
+	return cmd
+}
+
+// NewSystemInfoCommand prints system detail info
+func NewSystemInfoCommand(c common.Args) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "info",
+		Short: "Print the system deployment detail information in vela-system namespace.",
```

Comment:
Notice that while dealing with SystemInfo, the install namespace of KubeVela could be something other than "vela-system", so it would be better if we can have namespace-agnostic detection logics. This can be reserved as a future enhancement, not necessary for this PR.